### DISCUSSION TITLE
117

### DISCUSSION THREAD

**Discussion by giscus** [CODe5753/giscus#6] [Comments]
# 117

서론 회사에서 코드 리팩토링을 하는 업무를 하고 있었습니다. 레거시 코드 중 중첩 Map 형태로 반환하는 부분 때문에 프론트엔드에서는 불필요하게 반환된 객체의 key를 모두 읽고(Object.keys) 가장 최하단의 value에 직접 접근하는 로직을 항상 작성해야 했습니다. 그러다 보니 매번 프론트에서 작성되는 반복문 로직이 작성되어야만 했고 그 반복문 내에서 비즈니스 로직이라도 들어가게 되면 가독성을 매우 저하시키는 상황이 종종 생겼습니다. 그래서 이 중첩 Map 형태를 중첩이 없는 List형태로 반환하고자 코드를 수정하던 중, 누군가가 작성한 var 타입 추론을 만나게 됩니다. 간단히 눈으로 디버깅을 할 때 타입 추론은 IDE에 의존적임을 깨달았는데요, IDE가 없는 Github, GitLab 환경에..

https://imksh.com/117

**Comment by Hyeonqz** [comment]
좋은 내용 감사합니다!!

**Reply by CODe5753** [reply]
@Hyeonqz 안녕하세요!  
도움이 된 것 같아 기쁘네요!  
방문해 주셔서 감사합니다 :)

**Comment by mikekks** [comment]
저도 요즘 무분별하게 사용되는 var에 조금 부정적인 시각이었는데요!

그 이유에 대해 잘 정리해주셔서 저도 생각을 정리할 수 있었습니다 :)

저도 "진짜 필요한 상황이 아니면 쓰지 말아야겠다." 라는 생각으로 정리가 되는 것 같네요 !!
